packages/compiler/src/myst.ts

```diff
@@ -36,5 +59,95 @@ export function parsePanel(contents: string): Array<MystCard> {
     parsedCard.body = body;
     splitCards.push(parsedCard);
   }
-  return splitCards;
+
+  let myPanel: MystPanel = { cards: splitCards };
+  if (panelStyle) {
+    myPanel = { ...myPanel, style: panelStyle };
+  }
+  return myPanel;
+}
+
+export function parseStyling(contents: string): [string, string] {
+  let yamlBlock = "";
+  let returnContents = "";
+  const contentLines = contents.split("\n");
+  if (contents.startsWith(":")) {
+    const yamlLines: Array<string> = [];
+    while (contentLines) {
+      if (!ltrim(contentLines[0]).startsWith(":")) {
+        break;
+      } else if (contentLines.length > 0) {
+        // TODO: Find way to fix non-null assertion warning
+        yamlLines.push(ltrim(contentLines.shift()));
+      }
+    }
+    yamlBlock = yamlLines.join("\n");
+  }
+
+  returnContents = contentLines.join("\n").trim();
+  const returnPanel: [string, string] = [yamlBlock, returnContents];
+  return returnPanel;
+}
+
+export function parseYamlBlock(yamlBlock: string): Record<string, unknown> {
+  const styleContents = yamlBlock.split("\n");
+  const stylingRegex = /^:([a-z0-9]+):\s*([a-z\s\-0-9]+)/i;
+  // return object with key value pairs
+  let styles = {};
+  for (const line of styleContents) {
+    const stylingArray: RegExpExecArray | null = stylingRegex.exec(line);
+    if (stylingArray && stylingArray.length > 2) {
+      const key: string = stylingArray[1];
+      const value: string = stylingArray[2];
+      styles = { ...styles, [key]: value };
+    }
+  }
+  return styles;
+}
+
+function ltrim(rawString: string) {
+  return rawString.replace(/^\s+/gm, "");
+}
+
+export function mergeStyles(
+  initialStyle: MystStyling,
+  overridingStyle: MystStyling
+): MystStyling {
+  let k: keyof MystStyling;
+  for (k in initialStyle) {
+    if (k in overridingStyle && initialStyle[k] !== undefined && overridingStyle[k] !== undefined) {
```

Comment:
Wouldn't we want to use the values in `overridingStyle` even if `initialStyle` was undefined?

Reply:
This only iterates over the keys that are in the initialStyle, which may or may not be in the overriding style. With the current practice of mutating initialStyle it works as expected (since overridingStyle keys that are not in initialStyle keys are kept, but I will change to be in line with returning a new object instead.

Reply:
https://github.com/irydium/irydium/blob/ae9b5c1fc923c5c0f92b5ae15ed3782011f70ada/packages/compiler/src/myst.ts#L111